tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/NLog/LogsInjection/LogsInjectionHelper.cs

```diff
@@ -0,0 +1,227 @@
+// <copyright file="LogsInjectionHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.DirectSubmission.Proxies;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.DirectSubmission.Proxies.Pre43;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.LogsInjection
+{
+    /// <summary>
+    /// Helper class to add necessary configuration when logs injection is enabled.
+    /// </summary>
+    internal static class LogsInjectionHelper<TTarget>
+    {
+        private const string TraceId = "dd.trace_id";
+        private const string SpanId = "dd.span_id";
+        private const string Environment = "dd.env";
+        private const string Version = "dd.version";
+        private const string Service = "dd.service";
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(LogsInjectionHelper<TTarget>));
+        private static NLogVersion _version;
+        private static Type _jsonAttributeType;
+        private static Type _simpleLayoutType;
+
+        /// <summary>
+        ///     Adds necessary configuration elements to inject trace information in logs.
+        /// </summary>
+        /// <param name="loggingConfiguration">The NLog LoggingConfiguration to configure.</param>
+        public static void ConfigureLogsInjection(object loggingConfiguration)
+        {
+            if (loggingConfiguration == null)
+            {
+                return;
+            }
+
+            _version = NLogVersionHelper<TTarget>.Version;
+
+            if (!loggingConfiguration.TryDuckCast<IBasicLoggingConfigurationProxy>(out var loggingConfigurationProxy))
+            {
+                Log.Warning("Failed to DuckCast LoggingConfiguration");
```

Comment:
It should always work - at least it does for the supported versions 2+

Reply:
Would it be better to just remove the `TryDuckCast`? I never really know if I should always assume that these _could_ fail even if I don't think they can

Reply:
I think it depends tbh. The good thing about a failure in `DuckCast<>` is that exception bubbles all the way up to the instrumentation "machinery", we record it as a duck type failure, and it disables the integration (so we don't try again), and logs exactly what failed (e.g. a property missing, etc).

So if we _never_ expect it to fail, it's probably best to go with that IMO. `TryDuckCast` is particularly useful when we don't _know_ if it will be able to be duck typed, for example when the object could be one or the other, and you need to take two different courses of action. Or if it's not known in all code paths for example (like you had with the configuration sometimes being null etc or different versions)